crates/exex/src/manager.rs

```diff
@@ -318,18 +335,23 @@ impl Future for ExExManager {
                 None => return Err(()),
                 Some(height) => height,
             };
-
             if height < curr {
                 Ok(height)
             } else {
                 Ok(curr)
             }
         });
+
         if let Ok(finished_height) = finished_height {
             let _ = self.finished_height.send(FinishedHeight::Height(finished_height));
         }
 
-        Poll::Pending
+        // only return pending if the buffer is not empty
+        if !self.buffer.is_empty() {
+            return Poll::Pending;
+        } else {
+            return Poll::Ready(Ok(()));
+        }
```

Comment:
will this not immediately return `Poll::Pending` once the manager is spawned, since the buffer is empty from the beginning? i think what we want is to have the entire inner body of `poll` be in a `loop {}` and only return `Poll::Pending` if the buffer is not empty. we should never return `Poll::Ready(Ok(()))` since this terminates the manager

see the comments at the bottom of the manager file in https://github.com/paradigmxyz/reth/pull/7609